build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/views/endpoints_module.rb

```diff
@@ -9,13 +9,13 @@ def initialize(options)
 
         @parameters = @service.endpoint_rules.fetch('parameters', {})
 
-        @endpoint_classes = @service.api['operations'].each.with_object([]) do
-          |(name, op), array|
-          array << EndpointClass.new(
+        @endpoint_classes = @service.api['operations'].map do
+          |name, op|
+          EndpointClass.new(
             name: name,
             parameters: endpoint_parameters_for_operation(op)
           )
-        end
+        end.reject { |ec| ec.parameters.empty? }
```

Comment:
Ditto on my other comment. Could this just stay each with object and when you find empty params, skip item